src/examples/surgicalplanning/scene.js

```diff
@@ -0,0 +1,141 @@
+    spotLight.shadow.camera.near = 3;
+    spotLight.shadow.camera.far = 10;
+    spotLight.shadow.mapSize.width = 1024;
+    spotLight.shadow.mapSize.height = 1024;
+    scene.add( spotLight );
+
+    var dirLight = new DirectionalLight( 0x55505a, 1 );
+    dirLight.position.set( 0, 3, 0 );
+    dirLight.castShadow = true;
+    dirLight.shadow.camera.near = 1;
+    dirLight.shadow.camera.far = 10;
+
+    dirLight.shadow.camera.right = 1;
+    dirLight.shadow.camera.left = - 1;
+    dirLight.shadow.camera.top	= 1;
+    dirLight.shadow.camera.bottom = - 1;
+
+    dirLight.shadow.mapSize.width = 1024;
+    dirLight.shadow.mapSize.height = 1024;
+    scene.add( dirLight );
+
+    var localPlane = new Plane( new Vector3( 0, 0, 0.5 ), 0.8 );
+    Renderer.clippingPlanes = [localPlane]
+    Renderer.localClippingEnabled = true;
+
+    var clippingPlaneGeometry = new PlaneBufferGeometry( localPlane.x, localPlane.y, localPlane.z, 1 )
+    var clippingPlaneMaterial = new MeshBasicMaterial( {
+        color: 'green',
+        side: DoubleSide,
+
+        opacity: 0.2,
+        transparent: true,
+    } );
+
+    var clippingPlaneMesh = new Mesh( clippingPlaneGeometry, clippingPlaneMaterial );
+    clippingPlaneMesh.matrixAutoUpdate = false;
+
+    scene.add( clippingPlaneMesh );
+    clippingPlaneMesh.Update = () => {
+        //clippingPlaneMesh.set(localPlane.normal.x, localPlane.normal.y, localPlane.normal.z)
+    }
+
+    const loader = new GLTFLoader();
+
+    // "A simulated heart" developed by Ryan James
+    let heart;
+    loader.load(GLTFHeart, function (gltf)
+    {
+        heart = gltf.scene.children[ 0 ];
+        heart.position.set(0, 0, -0.5);
+        heart.scale.set(0.1, 0.1, 0.1);
+
+        networking.remoteSync.addSharedObject(heart);
+
+        var heartMaterial = new MeshPhongMaterial( {
+            color: 'red',
+            shininess: 100,
+            side: DoubleSide,
+
+            // ***** Clipping setup (material): *****
+            clippingPlanes: [ localPlane ],
+            clipShadows: true
+        });
+
+        heart.material = heartMaterial;
+
+        camera.add(heart);
+    });
+
+    const geo = new SphereBufferGeometry(.1, 13, 13);
+    const mat = new MeshNormalMaterial({ wireframe: true });
+    const mesh = new Mesh(geo, mat);
+
+    for(var gripIndex = 0; gripIndex < XRInput.controllerGrips.length; gripIndex++) {
+        var e = XRInput.controllerGrips[gripIndex];
+        const sp = mesh.clone();
+
+        sp.Update = () =>
+        {
+            sp.position.copy(e.position);
+            sp.quaternion.copy(e.quaternion);
+
+            if (e.isSelecting) {
+                heart.position.copy(e.position);
+                //heart.quaternion.copy(e.quaternion);
+            }
+
+            Renderer.xr.getController(gripIndex);
+        }
+        networking.remoteSync.addSharedObject(sp);
+        scene.add(sp);
+    }
+
+    State.eventHandler.addEventListener("xrsessionstarted", (e) =>
```

Comment:
Best way I could come up with to get the controllerGrip associated with the controller. It would be nice if there were a single object with both that I could use in the selectstart and selectend callbacks. That would allow me to override those methods on XRInput instead of duplicating the logic below, I think.

Reply:
Absolutely - let's go over latest State.eventHandler and XRInput hooks which, I think, better reflect the spec's intentions.